### PR TITLE
 add FleetAPI support Resnet50 and VGG16

### PR DESCRIPTION
**Pull request by danleifeng**
add FleetAPI support Resnet50 and VGG16:
- add models/vgg.py
- edit training shell script supporting multi-gpu
- add training_with_fleet.py with FleetAPI

### CODE REVIEW

File: benchmark/collective/resnet/utils/fp16_utils.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+from __future__ import print_function
```
Comment (guru4elephant) [inline benchmark/collective/resnet/utils/fp16_utils.py]: what's the different between this fp16_utils and https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/contrib/mixed_precision/fp16_utils.py

File: benchmark/collective/resnet/utils/img_tool.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+""" tools for processing images
```
Comment (guru4elephant) [inline benchmark/collective/resnet/utils/img_tool.py]: please add license here. and mark that the file is mainly from models.

File: benchmark/collective/resnet/utils/reader_cv2.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+""" tools for processing images
```
Comment (guru4elephant) [inline benchmark/collective/resnet/utils/reader_cv2.py]: please add license here.

Comment (danleifeng) [reply benchmark/collective/resnet/utils/reader_cv2.py]: Done.

File: benchmark/collective/resnet/utils/img_tool.py (hunk 2 above)
Comment (danleifeng) [reply benchmark/collective/resnet/utils/img_tool.py]: Done.

**Comment by danleifeng** [comment]
diffpr:http://gitlab.baidu.com/gongweibao/resnet/merge_requests/1

File: benchmark/collective/resnet/utils/fp16_utils.py (hunk 1 above)
Comment (danleifeng) [reply benchmark/collective/resnet/utils/fp16_utils.py]: the file is originally from paddle/model. The only difference is that the append_cast_op function is deleted and cast_fp16_to_fp32 and cast_fp32_to_fp16 functions are added.

**Pull request merged**